includes/EmailNotification.php

```diff
@@ -458,30 +455,68 @@ private function emailExtensionEnabled() {
 	}
 
 	/**
-	 * When a page is created, the previousRevId is always 0.
-	 * @return bool
-	 */
-	private function isNewPage() {
-		return $this->previousRevId == 0;
-	}
-
-	/**
 	 * Send a watched page edit email using the new Email extension.
 	 * @param $user User
 	 */
 	private function sendUsingEmailExtension( \User $user ) {
-		F::app()->sendRequest( 'Email\Controller\WatchedPage', 'handle',
-			[
+
+		if ( $this->isArticlePageEdit() ) {
+			$controller = 'Email\Controller\WatchedPage';
+		} elseif ( $this->isArticleComment() ) {
+			$controller = 'Email\Controller\ArticleComment';
+		} elseif ( $this->isBlogComment() ) {
+			$controller = 'Email\Controller\BlogComment';
```

Comment:
So this doesn't actually cover all cases. Adding a category also goes through this method (`$this->action === "categoryadd"`), as does the email for a create page (`empty( $this->action ) && $this->isNewPage();`) so those emails will get skipped by this current logic. Those are the only other 2 that I know of, but given the wilyness of MediaWiki, there could be others.

We should make sure that the category add and created page emails still flows through the `sendUsingUserMailer()` method, and also maybe log the $action when the controller is empty on L471 so we can see if we're missing anything else?